### PR TITLE
images: Add cockpit/base podman container

### PR DESCRIPTION
**Pull request by martinpitt**
On Fedora ≥ 31, use podman to build the cockpit/base container. This
will enable us to move the container/bastion test to Fedora 31.

Also make the "fedora:30" base image of cockpit/base dynamic, so that
building an image will automatically use a base image that corresponds
to the host Fedora version. With that we won't have to touch the
Dockerfile for future image upgrades any more, and the base container
will work on all our currently supported Fedora images.

Drop the long-obsolete dnf upgrade workaround, that bug was fixed in
2017.

 * [x] image-refresh fedora-31
 * [x] image-refresh fedora-30
 * [x] fix [check-metrics](https://logs.cockpit-project.org/logs/pull-562-20200224-172226-de3e8c58-fedora-31-cockpit-project-cockpit/log.html#125-2): https://github.com/cockpit-project/cockpit/pull/13608
 * [x] fix [selenium sosreport test](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-172225-de3e8c58-fedora-31-selenium-chrome-cockpit-project-cockpit/log.html#69): https://github.com/cockpit-project/cockpit/pull/13609
 * [x] fix selenium for rhel-8.2 branch: https://github.com/cockpit-project/cockpit/pull/13610
 * [x] fix [c-podman testBasicSystem](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html#1-2): https://github.com/cockpit-project/cockpit-podman/pull/322
 * [x] fix [c-podman testRunImageSystem](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html#7-2): https://github.com/cockpit-project/cockpit-podman/pull/323

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-562-20200224-075050/

**Comment by cockpituous** [comment]
image-refresh fedora-31 done: https://github.com/cockpituous/bots/commits/image-refresh-fedora-31-20200224-083131

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-02.
Log: https://logs.cockpit-project.org/logs/image-refresh-562-20200224-083131/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-562-20200224-083131/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-06.
Log: https://logs.cockpit-project.org/logs/image-refresh-562-20200224-091414/

**Comment by cockpituous** [comment]
image-refresh fedora-30 done: https://github.com/cockpituous/bots/commits/image-refresh-fedora-30-20200224-095454

**Comment by martinpitt** [comment]
[bastion fails on rhel-7.8 branch](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-102227-31ef5ea1-fedora-30-container-bastion-cockpit-project-cockpit-rhel-7.8/log.html). It already did that in #549, so this is unrelated to the changes here.

**Comment by martinpitt** [comment]
I figured out the c/bastion regression on rhel-7.8 branch. Fix pushed, have to rebuild f30 image.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-05.
Log: https://logs.cockpit-project.org/logs/image-refresh-562-20200224-120909/

**Comment by cockpituous** [comment]
image-refresh fedora-30 done: https://github.com/cockpituous/bots/commits/image-refresh-fedora-30-20200224-130707

File: images/scripts/fedora.setup [hunk 1]
```diff
@@ -200,6 +202,11 @@ fi
 # Configure kubernetes
 [ -z "$HAVE_KUBERNETES" ] || /var/lib/testvm/kubernetes.setup
 
+# for Fedora ≥ 31, use podman for cockpit/base container; we don't need kubernetes any more
```
Comment (marusak) [inline images/scripts/fedora.setup]: maybe stupid question, but what do we need `cockpit/base` for except for bastion testing?

Comment (martinpitt) [reply images/scripts/fedora.setup]: We also need it for testing cockpit/kubernetes on the rhel-7.8 branch. But on master, that's the remaining use case, yes. We could probably get rid of it and use the same approach as for cockpit/ws -- but that is based on the official container image, which doesn't exist yet for bastion.

**Comment by martinpitt** [comment]
The [podman failure](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html) looks a bit suspicious, I think it doesn't expect the new cockpit/base container?

**Comment by martinpitt** [comment]
The image refresh seems to break more stuff: [metrics](https://logs.cockpit-project.org/logs/pull-562-20200224-172226-de3e8c58-fedora-31-cockpit-project-cockpit/log.html#125-2) and [sosreport](https://logs.cockpit-project.org/logs/pull-562-20200224-172224-de3e8c58-fedora-31-selenium-firefox-cockpit-project-cockpit-rhel-8.2/log.html#69)

**Comment by martinpitt** [comment]
As for the podman regressions:
 * [testBasicSystem](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html#1-2) looks straightforward, it just doesn't seem to expect the cockpit/base container.
 * [testRunImageSystem](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html#7-2) looks like a podman change with CPU shares
 * [testRunImageUser](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-562-20200224-173233-de3e8c58-fedora-31-cockpit-project-cockpit-podman/log.html#8-2) looks strange -- it's waiting for some dialog to go away, but the screenshot doesn't show any dialog.

This fedora-31 update bumps podman to 1.8, so likely a few behaviour changes.

**Comment by martinpitt** [comment]
Last flake: [selenium/chrome on rhel-8.2 branch in MachinesStoragePoolTestSuite.testDeleteStoragePool](https://logs.cockpit-project.org/logs/pull-562-20200225-105820-de3e8c58-fedora-31-selenium-chrome-cockpit-project-cockpit-rhel-8.2/log.html). Retrying once more..

**Comment by martinpitt** [comment]
Now it failed in [firewall](https://logs.cockpit-project.org/logs/pull-562-20200225-123206-de3e8c58-fedora-31-selenium-chrome-cockpit-project-cockpit-rhel-8.2/log.html). Collectively, all tests passed now, so EOUTOFPATIENCE.

**Pull request merged**